src/components/dropdown/react/Dropdown.tsx

```diff
@@ -75,73 +76,68 @@ const DEFAULT_PROPS: IDefaultPropsType = {
 const Dropdown: React.FC<DropdownProps> = ({
     children,
     className = '',
+    onClose,
     closeOnClick = DEFAULT_PROPS.closeOnClick,
     escapeClose = DEFAULT_PROPS.escapeClose,
     offset,
-    position,
-    showDropdown = DEFAULT_PROPS.showDropdown,
-    toggleElement,
-    width,
+    showDropdown,
+    anchorRef,
+    placement = DEFAULT_PROPS.placement,
     ...props
 }: DropdownProps): React.ReactElement => {
     const wrapperRef: React.RefObject<HTMLDivElement> = useRef(null);
-    const [isOpen, setIsOpen]: [boolean, (isOpen: boolean) => void] = useState<boolean>(showDropdown || false);
-
-    function closeDropdown(): void {
-        setIsOpen(false);
-    }
-
-    function toggleDropdown(): void {
-        setIsOpen(!isOpen);
-    }
-
-    const anchorElement: React.ReactNode = (
-        <div className={`${CLASSNAME}__toggle`} onClick={toggleDropdown}>
-            {toggleElement}
-        </div>
+    const popoverRef: React.RefObject<HTMLDivElement> = useRef(null);
+
+    const { computedPosition, isVisible } = Popover.useComputePosition(
+        placement!,
+        anchorRef,
+        popoverRef,
+        showDropdown,
+        offset,
+        true,
     );
 
-    const popperElement: React.ReactNode = (
-        <div className={`${CLASSNAME}__menu`} style={{ width }}>
-            <div className={`${CLASSNAME}__content`}>{children(setIsOpen)}</div>
+    const popperElement: React.ReactElement = (
+        <div ref={wrapperRef} className={`${CLASSNAME}__menu`} style={{ width: computedPosition.width }}>
+            <div className={`${CLASSNAME}__content`}>{children}</div>
         </div>
     );
 
-    // Control the dropdown from the outside via the showDropdown prop.
-    useEffect((): void => {
-        if (showDropdown === undefined) {
-            return;
+    useEffect(() => {
+        if (escapeClose && showDropdown && wrapperRef.current) {
+            window.addEventListener('keydown', onEscapePressed(onClose!));
+        } else {
+            window.removeEventListener('keydown', onEscapePressed(onClose!));
         }
-
-        setIsOpen(showDropdown);
-    }, [showDropdown]);
+        return (): void => {
+            window.removeEventListener('keydown', onEscapePressed(onClose!));
+        };
+    }, [showDropdown, escapeClose]);
```

Comment:
The `onClose` is also a dependency of this `useEffect` and it might be `undefined` or not a function. Perhaps we could check it's value here before defining the `useEffect`?

Reply:
You're right